lib/elasticity/index_config.rb

```diff
@@ -20,7 +21,15 @@ def client
     end
 
     def definition
-      { settings: @elasticity_config.settings, mappings: { @document_type => @mapping } }
+      definition = { settings: @elasticity_config.settings, mappings: { } }
+      if subclasses.present?
+        subclasses.each do |subclass|
+          definition[:mappings][Object.const_get(subclass).document_type] = Object.const_get(subclass).mapping
+        end
+      else
+        definition[:mappings] = { @document_type => @mapping }
+      end
```

Comment:
From an API point of view there is nothing keeping you from specifying both a mapping in the parent class and a mapping on the subclass, so ideally we should merge the mapping + subclasses mappings rather than one or the others. What you think?